src/nn/ops/binary.ts

```diff
@@ -31,11 +31,11 @@ export abstract class Binary extends SingleOutputOperation {
         const rankA = a.rank;
         const rankB = b.rank;
         if (rankA === 1 && rankB === 1) {
-          outputShape = [];
-        } else if (rankA === 2 && rankB === 1) {
-          outputShape = [a.shape[0], 1];
-        } else if (rankA === 1 && rankB === 2) {
-          outputShape = [1, b.shape[1]];
+          outputShape = []; // scalar
+        } else if (rankA >= 2 && rankB === 1) {
+          outputShape = [...a.shape.slice(0, -1), 1];
+        } else if (rankA === 1 && rankB >= 2) {
+          outputShape = [...b.shape.slice(0, -2), 1, b.shape[1]];
```

Comment:
outputShape = [...b.shape.slice(0, -2), 1, b.shape[rankB-1]];?

Reply:
Yes, thanks for pointing this error out, I will update it.

Reply:
Updated, PTAL, thanks.